### ISSUE TITLE
Question of RAdam's dependence on the number of examples

### ISSUE THREAD

**Issue by brando90** [LiyuanLucasLiu/RAdam#64]
I'd like to confirm what:

> Specifically, we identify that, due to the limited amount of samples in the early stage of model training, the adaptive learning rate has an undesirably large variance and can cause the model to converge to suspicious/bad local optima.

means. I read the the paper and didn't see RAdam depend on the number of training examples. So I don't really undestand what you mean by " limited amount of samples in the early stage of model training". If you are training via epochs the model "sees" the entire training set at every "epoch step". So I am unsure how to interpret that. 

Thanks for sharing your work with us!

**Comment by LiyuanLucasLiu** [comment]
Hi, thanks for the question. 
The amount of samples here refer to the number of batches (e.g., if one epoch has 1000 batches,  the optimizer sees 1000 samples after its training on this epoch). I believe the controlled experiments (Adam-2k v.s. Adam wo warmup) is the best justification for this. Note that simply making the batch to be 2000x larger would still leads to convergence failure, if not using warmup. 

In RAdam, we try to update less when the variance is large (w. small number of batches).
